enhancements/cert-manager-proxy/http01_challenge_proxy.md

```diff
@@ -0,0 +1,107 @@
+
+However, cluster administrators often want to issue custom certificates for the API endpoint (`api.cluster.example.com`). Unlike other endpoints, this API endpoint is not exposed via the OpenShift Ingress. Depending on the OCP topology (e.g., SNO, MNO, Compact), it is exposed directly on the node or via a keepalive VIP. This lack of management by the OpenShift Ingress introduces challenges in obtaining certificates using an external ACME CA.
+
+The gap arises due to how the ACME HTTP01 challenge works. The following scenarios illustrate the challenges:
+
+1. **SNO (Single Node OpenShift)**: The same nodes host both the ingress and API components. Both FQDNs (`api` and wildcard) resolve to the same IP, making the challenge feasible.
+2. **Compact Clusters**: The node hosting the API VIP may also host an OpenShift Router. If no router is present on the node hosting the VIP, the challenge will fail.
+
+To address this gap, a small proxy was developed. This proxy runs on the cluster as a DaemonSet and ensures that connections reaching the API on port 80 are redirected to the OpenShift Ingress Routers. The proxy implementation uses `nftables` to redirect traffic from `API:80` to `PROXY:8888`.
+
+- **Proxy Code**: [GitHub Repository](https://github.com/mvazquezc/cert-mgr-http01-proxy/tree/main)
+- **Deployment Manifest**: [Manifest Link](https://github.com/mvazquezc/cert-mgr-http01-proxy/blob/main/manifests/deploy-in-ocp.yaml)
+
+This enhancement aims to provide a robust solution for managing certificates for the API endpoint in baremetal environments.
+
+### User Stories
+
+1. **As a cluster administrator**, I want to issue custom certificates for the API endpoint (`api.cluster.example.com`) using an external ACME CA, so that I can ensure secure communication for my cluster's API.
+2. **As a cluster administrator on a baremetal platform**, I want a reliable solution to handle HTTP01 challenges for the API endpoint, even when the endpoint is not managed by OpenShift Ingress, so that I can avoid manual workarounds.
+3. **As a developer**, I want a simple deployment mechanism for the HTTP01 challenge proxy, so that I can easily integrate it into my existing cluster setup.
+
+### Proposal
+
+The HTTP01 Challenge Proxy will be implemented via DaemonSet running on the cluster. It will:
+
+- Redirect HTTP traffic from the API endpoint (`api.cluster.example.com`) on port 80 to the OpenShift Ingress Routers.
+- Use `nftables` for traffic redirection from `API:80` to `PROXY:8888`.
+- Be deployed using a manifest that includes all necessary configurations.
+
+The proxy will ensure compatibility with various OCP topologies, including SNO, MNO, and Compact clusters, addressing the challenges of HTTP01 validation for the API endpoint.
+
+### API Extensions
+
+A new CR type may be created and can be applied to clusters.  This new typed will be stored in the [openshift/api](https://github.com/openshift/api) repo.
```

Comment:
Is a new CR required?

Reply:
This is something that I discussed as a possibility with @mvazquezc that it would be possibly a new CR that you would apply to get the proxy applied to the nodes.

Reply:
What group will this new CRD be created under? All new APIs will need to be tied to a TPNU feature-gate and start as `v1alpha1`.

What is this API going to look like?

Reply:
PR opened here: https://github.com/openshift/api/pull/2318

And I will add an example CR in this section.

Reply:
CR has been added.  I modified the `spec` section per a comment from @imiller0 a while ago, which basically makes it an empty section.

Reply:
@imiller0 @mvazquezc Correct me if I'm wrong but I believe this should be cluster wide correct?  I can adjust the CR in this case.

Reply:
+1 to cluster wide.